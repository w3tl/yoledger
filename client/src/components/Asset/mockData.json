[{
  "__typename": "Account",
  "id": "1",
  "name": "Bank Card",
  "balance": 0
}, {
  "__typename": "Account",
  "id": "2",
  "name": "Cash",
  "balance": 10
}, {
  "__typename": "Account",
  "id": "3",
  "name": "Bank Card 2",
  "balance": 150
}]
