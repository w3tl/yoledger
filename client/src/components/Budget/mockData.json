{
  "__typename": "BudgetPlan",
  "periods": ["2018-06-10", "2018-06-25", "2018-07-10", "2018-07-25", "2018-08-10", "2018-08-25"],
  "accounts": [{
    "account": {
      "__typename": "Account",
      "id": "1",
      "name": "Food"
    },
    "allocations": [{
      "__typename": "Allocation",
      "date": "2018-06-10",
      "amount": 10,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-06-25",
      "amount": 12,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-07-10",
      "amount": 14,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-07-25",
      "amount": 18,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-08-10",
      "amount": 7,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-08-25",
      "amount": 6,
      "balance": 0
    }]
  }]
}
