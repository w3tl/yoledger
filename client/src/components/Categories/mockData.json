{
  "expenses": [{
    "id": "21",
    "name": "Food",
    "type": "EXPENSE"
  }, {
    "id": "22",
    "name": "Beer",
    "type": "EXPENSE"
  }, {
    "id": "23",
    "name": "Train",
    "type": "EXPENSE"
  }],
  "incomes": [{
    "id": "31",
    "name": "Salary",
    "type": "INCOME"
  }, {
    "id": "32",
    "name": "Debts",
    "type": "INCOME"
  }, {
    "id": "33",
    "name": "Cashback",
    "type": "INCOME"
  }]
}
