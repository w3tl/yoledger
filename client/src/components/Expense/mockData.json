[{
  "id": "21",
  "name": "Food"
}, {
  "id": "22",
  "name": "Beer"
}, {
  "id": "23",
  "name": "Train"
}]
