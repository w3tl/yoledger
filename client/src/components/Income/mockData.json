[{
  "id": "31",
  "name": "Salary"
}, {
  "id": "32",
  "name": "Debts"
}, {
  "id": "33",
  "name": "Dividends"
}]
