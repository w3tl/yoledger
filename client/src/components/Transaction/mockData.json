[{
  "__typename": "Transaction",
  "id": "1",
  "amount": 10,
  "source": {
    "__typename": "Account",
    "id": "2",
    "name": "Cash"
  },
  "destination": {
    "__typename": "Account",
    "id": "21",
    "name": "Food"
  },
  "date": "2018-01-01T00:00:00.000Z"
}, {
  "__typename": "Transaction",
  "id": "2",
  "amount": 10,
  "source": {
    "__typename": "Account",
    "id": "2",
    "name": "Cash"
  },
  "destination": {
    "__typename": "Account",
    "id": "1",
    "name": "Bank Card"
  },
  "date": "2018-04-01T00:00:00.000Z"
}, {
  "__typename": "Transaction",
  "id": "3",
  "amount": 10,
  "source": {
    "__typename": "Account",
    "id": "1",
    "name": "Bank Card"
  },
  "destination": {
    "__typename": "Account",
    "id": "21",
    "name": "Food"
  },
  "date": "2018-01-06T00:00:00.000Z"
}]
