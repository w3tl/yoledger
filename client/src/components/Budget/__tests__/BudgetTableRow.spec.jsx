/* eslint-disable react/prop-types */
import React from 'react';
import { mount } from 'enzyme';
import RawBudgetTableRow from '../BudgetTableRow';
import { accounts } from '../mockData';

const periods = [new Date('2018-06-10'), new Date('2018-06-25'), new Date('2018-07-10')];

const BudgetTableRow = props => (
  <table>
    <tbody>
      <RawBudgetTableRow
        account={accounts[0]}
        periods={periods}
        {...props}
      />
    </tbody>
  </table>
);

describe('BudgetTableRow', () => {
  it('should call upsert', () => {
    const onUpdate = jest.fn();
    const wrapper = mount(<BudgetTableRow onUpdate={onUpdate} />);
    wrapper.find('BudgetTableCell').first().find('AmountInput').simulate('blur');
    expect(onUpdate.mock.calls[0]).toMatchSnapshot();
  });
});
