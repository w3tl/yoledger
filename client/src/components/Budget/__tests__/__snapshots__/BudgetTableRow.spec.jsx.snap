// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetTableRow should call upsert 1`] = `
Array [
  Object {
    "account": "Food",
    "date": 2018-06-10T00:00:00.000Z,
  },
]
`;
